homeassistant/components/sensor/opentherm_gw.py

```diff
@@ -0,0 +1,209 @@
+"""
+Support for OpenTherm Gateway sensors.
+
+For more details about this platform, please refer to the documentation at
+http://home-assistant.io/components/sensor.opentherm_gw/
+"""
+import logging
+
+from homeassistant.components.opentherm_gw import (
+    DATA_GW_VARS, DATA_OPENTHERM_GW, SIGNAL_OPENTHERM_GW_UPDATE)
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.const import DEVICE_CLASS_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+
+UNIT_BAR = 'bar'
+UNIT_HOUR = 'h'
+UNIT_KW = 'kW'
+UNIT_L_MIN = 'L/min'
+UNIT_PERCENT = '%'
+
+DEPENDENCIES = ['opentherm_gw']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the OpenTherm Gateway sensors."""
+    gw_vars = hass.data[DATA_OPENTHERM_GW][DATA_GW_VARS]
```

Comment:
Add a guard clause here that checks if discovery_info is None and if so returns.

Reply:
I will add the clause, but just for educational purposes: Can that actually happen? As I understand it, if the sensor is configured through the sensor component it will fail the config validation and otherwise it is only called from the opentherm_gw component with the required info. Is there another situation I am not aware of?

Reply:
Users can still add a section for the platform in the config yaml, which will trigger the platform to be set up that way besides being set up via discovery. That's what we are guarding from.

Reply:
Got it, thanks for the explanation.